git-repository/src/repository/state.rs

```diff
@@ -0,0 +1,42 @@
+use crate::RepositoryState;
+
+impl crate::Repository {
+    /// Returns the status of an in progress operation on a repository or [`RepositoryState::None`]
+    /// if nothing is happening.
+    pub fn in_progress_operation(&self) -> RepositoryState {
```

Comment:
I think this would be more idiomatic with `Option<RepositoryState>` while removing the `None` case from `RepositoryState`.